actionpack/lib/action_controller/metal/redirecting.rb

```diff
@@ -56,17 +60,14 @@ module Redirecting
     # To terminate the execution of the function immediately after the +redirect_to+, use return.
     #   redirect_to post_url(@post) and return
     def redirect_to(options = {}, response_options = {})
-      raise ActionControllerError.new("Cannot redirect to nil!") unless options
-      raise AbstractController::DoubleRenderError if response_body
+      response_options[:allow_other_host] ||= _allow_other_host unless response_options.key?(:allow_other_host)
 
-      self.status        = _extract_redirect_to_status(options, response_options)
-      self.location      = _compute_redirect_to_location(request, options)
-      self.response_body = "<html><body>You are being <a href=\"#{ERB::Util.unwrapped_html_escape(response.location)}\">redirected</a>.</body></html>"
+      _redirect_to(options, response_options)
```

Comment:
I think it would be cleaner if we inlined the logic from the `_redirect_to` method here and eliminated the private method.

Reply:
Agreed, I will revert this change for now.